sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,654 @@
+ * @param insel Input connection select - a MIO pad input to be connected to
+ * a peripheral input.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_select_result_t dif_pinmux_input_select(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t peripheral_input,
+    dif_pinmux_index_t insel);
+
+/**
+ * Sets a connection between a MIO pad output and peripheral output.
+ *
+ * A MIO pad output can be connected to any available peripheral output.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param mio_pad_output Padring MIO output index.
+ * @param outsel Peripheral output connection select - a Peripheral output to be
+ * connected to a MIO pad output.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_select_result_t dif_pinmux_output_select(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t mio_pad_output,
+    dif_pinmux_index_t outsel);
+
+/**
+ * Pin Multiplexer Padring pad attributes write error codes.
+ */
+typedef enum dif_pinmux_pad_write_attrs_result {
+  kDifPinmuxPadWriteAttrsOk = kDifPinmuxOk,
+  kDifPinmuxPadWriteAttrsError = kDifPinmuxError,
+  kDifPinmuxPadWriteAttrsBadArg = kDifPinmuxBadArg,
+  /**
+   * Peripheral is locked and attributes cannot be changed. Only a read has been
+   * performed and this error is recoverable but any operation that writes to
+   * the registers will fail.
+   */
+  kDifPinmuxPadWriteAttrsLocked,
+  /**
+   * Attribute change failed because it would result in conflicting attributes
+   * with another already enabled attribute. The change can be re-attempted with
+   * different bits. This could also occur if the attribute register already
+   * contained a conflicting set of attributes (e.g. because they have been
+   * written outside of the DIF with conflicting values).
+   */
+  kDifPinmuxPadWriteAttrsConflict,
+} dif_pinmux_pad_write_attrs_result_t;
+
+/**
+ * Writes attributes for a pad.
+ *
+ * This function completely overwrites the existing configuration of a pad, and
+ * has both enable and disable semantics.
+ *
+ * Not all pads implement all attributes and some combinations cannot be
+ * enabled together. This function returns a
+ * `kDifPinmuxPadWriteAttrsConflict` error in case of an invalid `attrs_in`.
+ * Conflicting attributes will be discarded by the hardware, and can be
+ * identified by comparing `attrs_in` to `attrs_out`.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Pad index to write the attributes for.
+ * @param type Pad type (MIO or DIO).
+ * @param attrs_in Attributes to write. A set bit writes an attribute, a
+ *                 clear bit leaves it in its existing state.
+ * @param attrs_out[out] The actual attributes written and accepted by the
+ *                       hardware.
+ * @return The result of the operation.
+ */
+dif_pinmux_pad_write_attrs_result_t dif_pinmux_pad_write_attrs(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t index,
```

Comment:
nit: Could call this `pad` rather than `index` here and elsewhere since the word index is already in the type name. Up to you.

Reply:
Done.